File: api/src/main/java/com/mystore/manager/api/common/payload/GlobalUserDatePayload.java
package com.mystore.manager.api.common.payload;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.Data;

@Data
@JsonIgnoreProperties(ignoreUnknown = true)
public class GlobalUserDatePayload extends GlobalDatePayload {
    private Integer createdById;
    private String createdByUsername;
    private String createdByFullName;
    private Integer updatedById;
    private String updatedByUsername;
    private String updatedByFullName;
    
    // POS information for multi-tenancy
    private Integer posId;
    private String posCode;
    private String posName;
}

File: ui/src/backend/payloads/global/GlobalUserDatePayload.ts
import { GlobalDatePayload } from './GlobalDatePayload';

export class GlobalUserDatePayload extends GlobalDatePayload {
  createdById: number | null;
  createdByUsername: string | null;
  createdByFullName: string | null;
  updatedById: number | null;
  updatedByUsername: string | null;
  updatedByFullName: string | null;

  constructor() {
    super();
    this.createdById = null;
    this.createdByUsername = null;
    this.createdByFullName = null;
    this.updatedById = null;
    this.updatedByUsername = null;
    this.updatedByFullName = null;
  }
}
